src/dmd/statement.h

```diff
@@ -129,21 +129,12 @@ class Statement : public ASTNode
     virtual Statements *flatten(Scope *sc);
     virtual Statement *last();
 
-    // Avoid dynamic_cast
-    virtual ErrorStatement *isErrorStatement() { return NULL; }
-    virtual ScopeStatement *isScopeStatement() { return NULL; }
-    virtual ExpStatement *isExpStatement() { return NULL; }
-    virtual CompoundStatement *isCompoundStatement() { return NULL; }
-    virtual ReturnStatement *isReturnStatement() { return NULL; }
-    virtual IfStatement *isIfStatement() { return NULL; }
-    virtual CaseStatement *isCaseStatement() { return NULL; }
-    virtual DefaultStatement *isDefaultStatement() { return NULL; }
-    virtual LabelStatement *isLabelStatement() { return NULL; }
-    virtual GotoDefaultStatement *isGotoDefaultStatement() { return NULL; }
-    virtual GotoCaseStatement *isGotoCaseStatement() { return NULL; }
-    virtual BreakStatement *isBreakStatement() { return NULL; }
-    virtual DtorExpStatement *isDtorExpStatement() { return NULL; }
-    virtual ForwardingStatement *isForwardingStatement() { return NULL; }
+    virtual ReturnStatement *endsWithReturnStatement() { return NULL; }
+
+    ExpStatement      *isExpStatement()     { return stmt == STMTexp      ? (ExpStatement*)this      : NULL; }
+    CompoundStatement *iCompoundStatement() { return stmt == STMTcompound ? (CompoundStatement*)this : NULL; }
+    ReturnStatement   *isReturnStatement()  { return stmt == STMTreturn   ? (ReturnStatement*)this   : NULL; }
+
```

Comment:
Why is it ok to remove the other stmt-types here?

Reply:
It's not OK to remove them, at least `isCaseStatement`, `isDefaultStatement`, and `isLabelStatement` are used by gdc.

Also, there's a spelling mistake.

Reply:
> Why is it ok

They're not worthy.